stlab/channel.hpp

```diff
@@ -176,6 +188,27 @@ auto get_process_state(const T& x) -> std::enable_if_t<!has_process_state<T>, pr
 /**************************************************************************************************/
 
 template <typename T>
+auto test_process_await_timeout(decltype(&T::await_timeout))->std::true_type;
+
+template <typename>
+auto test_process_await_timeout(...)->std::false_type;
+
+template <typename T>
+constexpr bool has_process_await_timeout = decltype(test_process_await_timeout<T>(0))::value;
+
+template <typename T>
+auto get_process_await_timeout(const T& x) -> std::enable_if_t<has_process_await_timeout<T>, std::chrono::milliseconds> {
```

Comment:
I think the await timeout should be part of the state callback instead of an additional callback. Return a pair and require (for now) a timeout of 0 if in a yield state (we likely want to support a timer on yield in the future to throttle output).